engine/server/sv_client.c

```diff
@@ -2091,6 +2091,744 @@ static qboolean SV_SendBuildInfo_f( sv_client_t *cl )
+		ent = SV_GetCrossEnt(cl->edict);
+
+		if (!SV_IsValidEdict(ent))
+			return NULL;
+
+		i = NUM_FOR_EDICT(ent);
+	}
+	else if (pattern[0] == '!') // Check for correct instanse with !(num)_(serial)
+	{
+		const char* p = pattern + 1;
+		i = Q_atoi(p);
+
+		while (isdigit(*p)) p++;
+
+		if (*p++ != '_')
+			return NULL;
+
+		if (i >= svgame.numEntities)
+			return NULL;
+
+		ent = EDICT_NUM(i);
+
+		if (ent->serialnumber != Q_atoi(p))
+			return NULL;
+	}
+	else
+	{
+		for (i = svgame.globals->maxClients + 1; i < svgame.numEntities; i++)
+		{
+			ent = EDICT_NUM(i);
+
+			if (!SV_IsValidEdict(ent))
+				continue;
+
+			if (Q_stricmpext(pattern, STRING(ent->v.targetname)))
+				break;
+		}
+	}
+
+	ent = EDICT_NUM(i);
+
+	if (!SV_IsValidEdict(ent))
+		return NULL;
+
+	return ent;
+}
+
+
+/*
+===============
+SV_EntInfo_f
+Print specified entity information to client
+===============
+*/
+void SV_EntInfo_f(sv_client_t* cl)
+{
+	edict_t* ent = NULL;
+	vec3_t borigin;
+
+	if (Cmd_Argc() != 2)
+	{
+		SV_ClientPrintf(cl, "Use ent_info <index|name|inst>\n");
+		return;
+	}
+
+	ent = SV_EntFindSingle(cl, Cmd_Argv(1));
+
+	if (!SV_IsValidEdict(ent)) return;
+
+	VectorAdd(ent->v.absmin, ent->v.absmax, borigin);
+	VectorScale(borigin, 0.5, borigin);
+
+	SV_ClientPrintf(cl, "origin: %.f %.f %.f\n", ent->v.origin[0], ent->v.origin[1], ent->v.origin[2]);
+
+	SV_ClientPrintf(cl, "angles: %.f %.f %.f\n", ent->v.angles[0], ent->v.angles[1], ent->v.angles[2]);
+
+	SV_ClientPrintf(cl, "borigin: %.f %.f %.f\n", borigin[0], borigin[1], borigin[2]);
+
+	if (ent->v.classname)
+		SV_ClientPrintf(cl, "class: %s\n", STRING(ent->v.classname));
+
+	if (ent->v.globalname)
+		SV_ClientPrintf(cl, "global: %s\n", STRING(ent->v.globalname));
+
+	if (ent->v.targetname)
+		SV_ClientPrintf(cl, "name: %s\n", STRING(ent->v.targetname));
+
+	if (ent->v.target)
+		SV_ClientPrintf(cl, "target: %s\n", STRING(ent->v.target));
+
+	if (ent->v.model)
+		SV_ClientPrintf(cl, "model: %s\n", STRING(ent->v.model));
+
+	SV_ClientPrintf(cl, "health: %.f\n", ent->v.health);
+
+	if (ent->v.gravity != 1.0f)
+		SV_ClientPrintf(cl, "gravity: %.2f\n", ent->v.gravity);
+
+	SV_ClientPrintf(cl, "movetype: %d\n", ent->v.movetype);
+
+	SV_ClientPrintf(cl, "rendermode: %d\n", ent->v.rendermode);
+	SV_ClientPrintf(cl, "renderfx: %d\n", ent->v.renderfx);
+	SV_ClientPrintf(cl, "renderamt: %f\n", ent->v.renderamt);
+	SV_ClientPrintf(cl, "rendercolor: %f %f %f\n", ent->v.rendercolor[0], ent->v.rendercolor[1], ent->v.rendercolor[2]);
+
+	SV_ClientPrintf(cl, "maxspeed: %f\n", ent->v.maxspeed);
+
+	if (ent->v.solid)
+		SV_ClientPrintf(cl, "solid: %d\n", ent->v.solid);
+	SV_ClientPrintf(cl, "flags: 0x%x\n", ent->v.flags);
+	SV_ClientPrintf(cl, "spawnflags: 0x%x\n", ent->v.spawnflags);
+}
+
+void SV_EntSendVars(sv_client_t* cl, edict_t* ent)
+{
+	if (!ent)
+		return;
+	MSG_WriteByte(&cl->netchan.message, svc_stufftext);
+	MSG_WriteString(&cl->netchan.message, va("set ent_last_name \"%s\"\n", STRING(ent->v.targetname)));
```

Comment:
Something tells me we don't have "set" command anymore. It's have to be restored.

Reply:
It seems that this method is useless so I just deleted it.

Reply:
It isn't?

The idea is to send new entity parameters like name, origin, class, etc so it can be used in callback alias.

Like this:

```
alias ent_last_cb "echo New entity origin is $ent_last_origin"
ent_create monster_scientist
```

It will print entity origin. AFAIK it's already used in some simple mods.